pkg/bgpv1/gobgp/server.go

```diff
@@ -117,39 +117,58 @@ func (g *GoBGPServer) AddNeighbor(ctx context.Context, n types.NeighborRequest)
 
 // UpdateNeighbor will update the existing CiliumBGPNeighbor in the gobgp.BgpServer.
 func (g *GoBGPServer) UpdateNeighbor(ctx context.Context, n types.NeighborRequest) error {
-	peer, err := g.getPeerConfig(ctx, n.Neighbor, true)
+	peer, needsFullReset, err := g.getPeerConfig(ctx, n.Neighbor, true)
```

Comment:
Nit: I'd prefer calling this "hard reset" since we are using the term "soft reset" already.